engine/server/engine/main.go

```diff
@@ -131,6 +134,40 @@ func runMain() error {
 		},
 	)
 
+	go func() {
+		staticPath := "/run/webapp"
+		indexPath := "index.html"
+
+		fileServer := http.FileServer(http.Dir(staticPath))
+
+		handler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			path, err := filepath.Abs(r.URL.Path)
+			if err != nil {
+				http.Error(w, err.Error(), http.StatusBadRequest)
+				return
+			}
+
+			path = filepath.Join(staticPath, path)
+
+			_, err = os.Stat(path)
+			if os.IsNotExist(err) {
+				// file does not exist, serve index.html
+				http.ServeFile(w, r, filepath.Join(staticPath, indexPath))
+				return
+			} else if err != nil {
+				http.Error(w, err.Error(), http.StatusInternalServerError)
+				return
+			}
+
+			fileServer.ServeHTTP(w, r)
+		})
+
+		err := http.ListenAndServe(webappPortAddr, handler)
+		if err != nil {
+			logrus.Debugf("error while starting the webapp: \n%v", err)
```

Comment:
if something fails here the engine will start without the HTTP server, I think it should fail loudly if the HTTP server can't start. So, you can pass an error channel to this subroutine and wait (in the main thread) until there is no error in order to continue

Reply:
i don't think so i need to this atm because it's a prototype and it shouldn't impact the normal flow. When we productize fully, I agree we can fail loudly.